src/main/scala/dpla/ingestion3/mappers/providers/MtMapping.scala

```diff
@@ -0,0 +1,169 @@
+package dpla.ingestion3.mappers.providers
+
+import java.net.URI
+
+import dpla.ingestion3.enrichments.normalizations.StringNormalizationUtils._
+import dpla.ingestion3.enrichments.normalizations.filters.{DigitalSurrogateBlockList, ExtentIdentificationList, FormatTypeValuesBlockList}
+import dpla.ingestion3.mappers.utils.{Document, IdMinter, Mapping, XmlExtractor}
+import dpla.ingestion3.messages.{IngestMessage, IngestMessageTemplates, MessageCollector}
+import dpla.ingestion3.model.DplaMapData.{AtLeastOne, ExactlyOne, ZeroToMany, ZeroToOne}
+import dpla.ingestion3.model._
+import dpla.ingestion3.utils.Utils
+import org.json4s.JValue
+import org.json4s.JsonDSL._
+
+import scala.util.{Failure, Success, Try}
+import scala.xml._
+
+
+class MtMapping extends Mapping[NodeSeq] with XmlExtractor with IdMinter[NodeSeq] with IngestMessageTemplates {
+
+  val formatBlockList: Set[String] =
+    DigitalSurrogateBlockList.termList ++
+      ExtentIdentificationList.termList
+
+  // ID minting functions
+  override def useProviderName(): Boolean = true
+
+  override def getProviderName(): String = "mt"
+
+  override def getProviderId(implicit data: Document[NodeSeq]): String =
+    extractString(data \\ "header" \ "identifier").getOrElse(throw new RuntimeException(s"No ID for record $data"))
+
+  // SourceResource mapping
+  override def collection(data: Document[NodeSeq]): Seq[DcmiTypeCollection] =
+    // first instance of <mods:relatedItem><mods:titleInfo><mods:title>
+    extractStrings(data \\ "relatedItem" \ "titleInfo" \ "title")
+      .map(nameOnlyCollection)
+      .take(1)
+
+  override def creator(data: Document[NodeSeq]): Seq[EdmAgent] =
+    // <mods:name><mods:namePart> when <mods:role><mods:roleTerm> equals Creator
+  (data \\ "name")
+    .map(node => node.filter(n => (n \\ "roleTerm").text.equalsIgnoreCase("creator")))
+    .flatMap(n => extractStrings(n \ "namePart"))
+    .map(nameOnlyAgent)
+
+  override def date(data: Document[NodeSeq]): Seq[EdmTimeSpan] =
+  // <mods:originInfo><mods:dateCreated>
+    extractStrings(data \\ "originInfo" \ "dateCreated")
+      .map(stringOnlyTimeSpan)
+
+  override def description(data: Document[NodeSeq]): Seq[String] =
+  // <mods:note> @type=content
+    (data \\ "metadata" \ "mods" \ "note")
+      .flatMap(n => getByAttribute(n.asInstanceOf[Elem], "type", "content"))
+      .flatMap(n => extractStrings(n))
+
+  override def extent(data: Document[NodeSeq]): ZeroToMany[String] =
```

Comment:
I notice that you specify a return type of `ZeroToMany[String]` here but in other places, such as on line 61, you specify a return type of `Seq[String]`.  Is there a rule of thumb for when to use the custom types?

Reply:
That was just the IntelliJ auto complete function filling in `Seq` instead of `ZeroToMany`